### PR TITLE
✨ Unify configuration, expose database mode setting via dotenv

### PR DESCRIPTION
**Pull request by muchzill4**
### Description

Unifies the way we configure parameters of `mbed-targets` - via `.env` file.

Please review the documentation output on azure.

### Test Coverage

- [x]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: mbed_targets/mbed_targets.py [hunk 1]
```diff
@@ -181,15 +164,13 @@ def json_dump(self) -> str:
         return json.dumps([asdict(t) for t in self], indent=4)
 
 
-def _get_target(query: TargetDatabaseQuery, mode: DatabaseMode = DatabaseMode.AUTO) -> MbedTarget:
-    if mode == DatabaseMode.OFFLINE:
+def _get_target(query: TargetDatabaseQuery) -> MbedTarget:
+    if MBED_DATABASE_MODE == DatabaseMode.OFFLINE:
         return MbedTargets.from_offline_database().get_target(**query)
-    if mode == DatabaseMode.ONLINE:
+    elif MBED_DATABASE_MODE == DatabaseMode.ONLINE:
```
Comment (rwalton-arm) [inline mbed_targets/mbed_targets.py]: Why do we need `elif` here? Another thing I noticed which isn't part of this PR, we have a redundant `else` clause lower down.

File: tests/test_target_database.py [hunk 2]
```diff
@@ -48,26 +48,26 @@ def test_200_value_data(self, mock_request):
...
+        os.environ["MBED_API_AUTH_TOKEN"] = ""
+        target_database._get_request()
+        requests.get.assert_called_once_with(target_database._TARGET_API, headers=None)
 
-    def test_raises_tools_error_on_connection_error(self):
-        with mock.patch("mbed_targets._internal.target_database.requests.get") as mock_req_get:
-            mock_req_get.side_effect = target_database.requests.exceptions.ConnectionError
-            with self.assertRaises(target_database.TargetAPIError):
-                target_database._get_request()
+    @mock.patch("mbed_targets._internal.target_database.requests.get")
```
Comment (rwalton-arm) [inline tests/test_target_database.py]: Nice change.

File: mbed_targets/mbed_targets.py (hunk 1 above)
Comment (muchzill4) [reply mbed_targets/mbed_targets.py]: Am I correct in understanding that you're suggesting:
```
if a:
  return ..
if b:
  return ...
return ...
```
?

File: mbed_targets/__init__.py [hunk 3]
```diff
@@ -2,39 +2,60 @@
...
-for a private target, mbed-targets will need to contact the online database.
+Configuration
+=============
 
-To fetch data about private targets from the online database, the user must have an account
-on os.mbed.com and be member of a vendor team that has permissions to see the private board.
-An authentication token for the team member must be provided in an environment variable named
-`MBED_API_AUTH_TOKEN`. The package also accepts a `.env` file containing the environment
-variable definition as follows:
+How to configure mbed-targets?
```
[the remark this answers is not in the document]
Comment (muchzill4) [reply mbed_targets/__init__.py]: Unless you suggest 2 line change I'll need to fix this offline as the underline needs to match the length of the sub-heading afaik. 🤷‍♂

File: mbed_targets/mbed_targets.py (hunk 1 above)
Comment (rwalton-arm) [reply mbed_targets/mbed_targets.py]: I am. We return in each branch so we don't need `elif` or `else`

File: mbed_targets/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rwalton-arm) [reply mbed_targets/__init__.py]: Is it markdown? I don't think the length needs to match, although if it does that's super-annoying.

File: mbed_targets/mbed_targets.py (hunk 1 above)
Comment (muchzill4) [reply mbed_targets/mbed_targets.py]: https://github.com/ARMmbed/mbed-targets/pull/102/commits/5f6f6f9c3a1285ded2b8992d47b38220f171ba6c

File: mbed_targets/__init__.py [hunk 4]
```diff
@@ -2,39 +2,60 @@
...
-MBED_API_AUTH_TOKEN=token
+VARIABLE=value
 ```
 
-Where `token` is the authentication token.
+The `.env` file takes precendence, meaning the values set in the file will override any
+values previously set in your environment.
+
+.. WARNING::
+   Do not upload `.env` files containing private tokens to GitHub! If you use this package
```
[the remark this answers is not in the document]
Comment (muchzill4) [reply mbed_targets/__init__.py]: https://github.com/ARMmbed/mbed-targets/pull/102/commits/5757784a4f98c04eab19f87d922593c92cb13dfd

File: mbed_targets/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (muchzill4) [reply mbed_targets/__init__.py]: Oh, you're right. I must've done something wrong previously as I was sure it didn't work with an underline that didn't match the length of the heading. 👍

[the remark this answers is not in the document]
Comment (muchzill4) [reply mbed_targets/__init__.py]: https://github.com/ARMmbed/mbed-targets/pull/102/commits/7136cf535be8ec28ceff782feda4db35d0d1ec44 - adjusted underline too as I have ocd.

**Pull request merged**